### PR TITLE
[ROT-374]: Build-Your-Plate E2E Tests

### PR DESCRIPTION
**Pull request by RobertSaylesAND**

### CODE REVIEW

**Review by AshleyBeel — APPROVED** [review]

File: cypress/e2e/build-your-plate.cy.ts [hunk 1]
```diff
@@ -1,6 +1,257 @@
...
 describe("Build Your Plate", () => {
-  it("gives correct score with 5 sweets");
-  it("gives correct score with 5 meats");
-  it("gives correct score with 5 mixed items");
-  it("allows user to swap food items before scoring");
+  beforeEach(() => {
+    cy.visit("http://localhost:3000/Games/BuildYourPlate");
+  });
+
+  it("check a11y buuld-your-plate page", () => {
```
Comment (ALougherANDDigital) [inline cypress/e2e/build-your-plate.cy.ts:9]: Typo - "buuld" instead of build

File: cypress/e2e/build-your-plate.cy.ts [hunk 2]
```diff
@@ -1,6 +1,257 @@
...
+    cy.visit("http://localhost:3000/");
+    cy.get(".react-multi-carousel-list [data-testid='Games']").click();
+    cy.get("[data-testid='menu-styling'] [data-testid='Build your plate']").click();
+    cy.url().should("eq", "http://localhost:3000/Games/BuildYourPlate");
+  });
+
+  it("gives correct score with 5 sweets", () => {
+    cy.get("[data-testid='game-begin']").click();
+    cy.contains("button", "Play").click();
+    cy.contains("button", "Score my plate").should("be.disabled");
```
Comment (ALougherANDDigital) [inline cypress/e2e/build-your-plate.cy.ts:24]: The Score my plate button being disabled prior to any selections could be its own short test.

File: cypress/e2e/build-your-plate.cy.ts [hunk 3]
```diff
@@ -1,6 +1,257 @@
...
+
+    cy.get("[alt='Tick']").should("have.length", 5);
+    cy.contains("button", "Score my plate").should("be.enabled");
+    cy.contains("5 / 5");
+    cy.contains("button", "Score my plate").click();
+    cy.get("[alt='plate']");
+    cy.contains("button", "Score my plate").click();
+    cy.contains("Score: 10 out of 50");
+    cy.contains("Could be better!");
+    cy.contains("button", "Play again").click();
```
Comment (ALougherANDDigital) [inline cypress/e2e/build-your-plate.cy.ts:44]: No need to click play again at the end of the test unless specifically testing for the functionality to play again

File: cypress/e2e/build-your-plate.cy.ts [hunk 4]
```diff
@@ -1,6 +1,257 @@
...
+    cy.contains("button", "Score my plate").should("be.enabled");
+    cy.contains("5 / 5");
+    cy.contains("button", "Score my plate").click();
+    cy.get("[alt='plate']");
+    cy.contains("button", "Score my plate").click();
+    cy.get<number[]>("@expectedScore").then((score) => {
+      cy.contains(`Score: ${score[0]} out of 50`);
+    });
+    cy.contains("Great!");
+    cy.contains("button", "Play again").click();
```
Comment (ALougherANDDigital) [inline cypress/e2e/build-your-plate.cy.ts:77]: Same as prior comment regarding play again button

File: cypress/e2e/build-your-plate.cy.ts [hunk 5]
```diff
@@ -1,6 +1,257 @@
...
+  });
+
+  it("gives correct score with 5 mixed items", () => {
```
Comment (ALougherANDDigital) [inline cypress/e2e/build-your-plate.cy.ts:80]: Unsure about having a test with a randomised output

File: cypress/e2e/build-your-plate.cy.ts [hunk 6]
```diff
@@ -1,6 +1,257 @@
...
+    cy.contains("5 / 5");
+    cy.contains("button", "Score my plate").click();
+    cy.get("[alt='plate']");
+    cy.contains("button", "Score my plate").click();
+    cy.contains("Score: 10 out of 50");
+    cy.contains("Could be better!");
+    cy.contains("button", "Play again").click();
+  });
+
+  it("gives correct score with 5 meats", () => {
```
Comment (ALougherANDDigital) [inline cypress/e2e/build-your-plate.cy.ts:47]: Unsure if testing 5 meats after 5 sweets is necessary as not testing different functionality.
Might be more worthwhile to have a test for 5 of the same category and 5 of different mixed categories.

**Review by ALougherANDDigital — CHANGES_REQUESTED** [review]

File: cypress/e2e/build-your-plate.cy.ts (hunk 5 above)
Comment (RobertSaylesAND) [reply cypress/e2e/build-your-plate.cy.ts:47]: Note - discussed this briefly with Andrew and we've agreed that the current approach is our best option at present. While there may be alternative options for handling the randomness, we don't have one that would be better, either requiring substantial reworking beyond the scope of this ticket or requiring methods previously discussed as impractical.

**Pull request merged**